Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -1458,16 +1458,25 @@ private extension ChatChannelController {
     }
 
     private func getFirstUnreadMessageId() -> MessageId? {
-        guard let lastReadMessageId = channel?.reads.first(where: { $0.user.id == client.currentUserId })?.lastReadMessageId else {
-            // We default to the oldest message in the history
-            return messages.last?.id
+        let messages = self.messages.filter {
```

Comment:
Why is this still here?

Reply:
The messages are used in multiple places, it is not enough, change it it only in `messages.last?.id`. Since we use `firstIndex` as well, we can't assume we will only need the first or last message 🤔